### ISSUE TITLE
Checkbox/Radio - conditional input focus on click

### ISSUE THREAD

**Issue by marianlorinc** [primefaces/primereact#2280]
### For RadioButton and Checkbox component

Because checkbox input is still focused again and again after each click, I can not style .p-checkbox-focused class properly only for "focus-visible" state. If you add a new option to checkbox props, something like a blurOnClick:boolean, and use this condition in onClick function for input.focus(), then developer can achieve "focus-visible" styling only for keyboard navigation to checkbox.

NOTICE! I need to style the highest wrapper, the element with .p-checkbox class, not input itself. And it is not possible to do only with :focus-visible selector in css.

```
    onClick(e) {
        if (!this.props.disabled && !this.props.readOnly && this.props.onChange) {
            this.props.onChange({
                originalEvent: e,
                value: this.props.value,
                checked: !this.props.checked,
                stopPropagation : () =>{},
                preventDefault : () =>{},
                target: {
                    type: 'checkbox',
                    name: this.props.name,
                    id: this.props.id,
                    value: this.props.value,
                    checked: !this.props.checked,
                }
            });

            this.inputRef.current.checked = !this.props.checked;
            this.inputRef.current.focus(); //TO FIX: condition should be here !!!

            e.preventDefault();
        }
    }
```

**Current behavior**
.p-checkbox-focused is add to Checkbox classlist always, for each click and there is no option to recognize focus and focus-visible state.

**Expected behavior**
Make focusing on each mouse click as optional, or switchable feature via props.
What I recommend:
```
onClick(e) {
        if (!this.props.disabled && !this.props.readOnly && this.props.onChange) {
            this.props.onChange({
                originalEvent: e,
                value: this.props.value,
                checked: !this.props.checked,
                stopPropagation: () => { },
                preventDefault: () => { },
                target: {
                    type: 'checkbox',
                    name: this.props.name,
                    id: this.props.id,
                    value: this.props.value,
                    checked: !this.props.checked,
                }
            });

            this.inputRef.current.checked = !this.props.checked;

            const isEnterPressed = e.type === 'keydown' && e.keyCode === 13;
            if (!isEnterPressed  && this.props.blurOnClick) {
                this.inputRef.current.blur();
            } else {
                this.inputRef.current.focus();
            }

            e.preventDefault();
        }
    }
```

**Minimal reproduction of the problem with instructions**

**Please tell us about your environment:**

* **React version:**
^17.0.2

* **PrimeReact version:**
^6.5.0

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]
all
 
* **Language:** [all | TypeScript X.X | ES6/7 | ES5]

**Comment by habubey** [comment]
Issue tracker is used for defects only as part of our commitment to quality and continuous improvement in all areas. Enhancements are collected as valuable community feedback and managed internally so moving this enhancement ticket to our internal project management backlog.
